src/lib/hooks/useConnect/setupConnection.ts

```diff
@@ -82,126 +66,190 @@ export async function setUpConnection({
-            });
-          });
-
-        // use nickname/profileUrl if provided
-        // or set userID as nickname
-        if ((nickname !== user.nickname || profileUrl !== user.profileUrl)
-          && !(isTextuallyNull(nickname) && isTextuallyNull(profileUrl))
-        ) {
-          logger?.info?.('SendbirdProvider | useConnect/setupConnection/updateCurrentUserInfo', {
-            nickname,
-            profileUrl,
-          });
-          newSdk.updateCurrentUserInfo({
-            nickname: nickname || user.nickname || (isUserIdUsedForNickname ? user.userId : ''),
-            profileUrl: profileUrl || user.profileUrl,
-          }).then((namedUser) => {
-            logger?.info?.('SendbirdProvider | useConnect/setupConnection/updateCurrentUserInfo success', {
-              nickname,
-              profileUrl,
-            });
-            userDispatcher({ type: UPDATE_USER_INFO, payload: namedUser });
-          }).finally(() => {
-            resolve();
-          });
-        } else {
-          resolve();
+        try {
+          // use nickname/profileUrl if provided or set userID as nickname
+          if (
+            (nickname !== user.nickname || profileUrl !== user.profileUrl)
+            && !(isTextuallyNull(nickname) && isTextuallyNull(profileUrl))
+          ) {
+            logger.info?.('SendbirdProvider | useConnect/setupConnection/updateCurrentUserInfo', { nickname, profileUrl });
+            const updateParams = {
+              nickname: nickname || user.nickname || (isUserIdUsedForNickname ? user.userId : ''),
+              profileUrl: profileUrl || user.profileUrl,
+            };
+
+            const updatedUser = await sdk.updateCurrentUserInfo(updateParams);
+            logger.info?.('SendbirdProvider | useConnect/setupConnection/updateCurrentUserInfo success', updateParams);
+            userDispatcher({ type: UPDATE_USER_INFO, payload: updatedUser });
+          }
+        } catch {
+          // NO-OP
         }
+
+        resolve();
+        eventHandlers?.connection?.onConnected?.(user);
       };
 
-      const connectCbError = (e: SendbirdError) => {
+      const onConnectFailed = async (e: SendbirdError) => {
+        if (sdk.isCacheEnabled && shouldClearCache(e)) {
+          logger.error?.(`SendbirdProvider | useConnect/setupConnection/connect clear cache [${e.code}/${e.message}]`);
+          await sdk.clearCachedData();
+        }
+
         const errorMessage = getConnectSbError(e);
-        logger?.error?.(errorMessage, {
-          e,
-          appId,
-          userId,
-        });
-        sdkDispatcher({ type: RESET_SDK });
+        logger.error?.(errorMessage, { e, appId, userId });
         userDispatcher({ type: RESET_USER });
-
+        sdkDispatcher({ type: RESET_SDK });
         sdkDispatcher({ type: SDK_ERROR });
-        onConnectionFailed?.(e);
-        // exit promise with error
+
         reject(errorMessage);
+        eventHandlers?.connection?.onFailed?.(e);
```

Comment:
shouldn't reject be called at the last? or it does not matter?

Reply:
It doesn’t really matter